scripts/diff.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# 現在のブランチを取得
+current_branch=$(git branch --show-current)
+
+# Issue番号の特定（ブランチ名から抽出）
+issue=$(echo "$current_branch" | grep -oE 'GH-[0-9]+' | grep -oE '[0-9]+')
+
+# 基準ブランチ（デフォルトはmain）
+base_branch=${1:-"main"}
+
+# main ブランチを最新化
+if [ "$base_branch" = "main" ]; then
+  git fetch origin main:main 2>/dev/null || true
+fi
+
+# 現在のブランチと基準ブランチの共通のマージベースを確認
+merge_base=$(git merge-base $base_branch $current_branch 2>/dev/null)
+
+if [ -n "$merge_base" ]; then
+  # 基準ブランチと現在のブランチに共通のマージベースが存在する場合
+  latest_base=$base_branch
+  latest_commit=$merge_base
+else
+  
+  # 派生元ブランチの特定（ローカル全ブランチから自動判定）
+  candidates=($(git branch --format='%(refname:short)' | grep -v "^$current_branch$"))
+  latest_base=""
+  latest_commit=""
+  for b in "${candidates[@]}"; do
+    commit=$(git merge-base $b $current_branch 2>/dev/null)
+    if [ -n "$commit" ]; then
+      if [ -z "$latest_commit" ] || [ "$(git log -1 --format=%ct $commit)" -gt "$(git log -1 --format=%ct $latest_commit)" ]; then
+        latest_commit=$commit
+        latest_base=$b
+      fi
+    fi
+  done
+  
+  if [ -n "$latest_base" ]; then
+    if [ "$latest_base" != "main" ] && git show-ref --verify --quiet refs/heads/main; then
+      # 派生元ブランチの最新の共通コミットが main に取り込まれているか確認
+      if git branch --contains $latest_commit | grep -q "main"; then
+        latest_base="main"
```

Comment:
![must-badge](https://img.shields.io/badge/review-must-red.svg)

ハードコーディングされているため、`base_branch`を参照するようにしてください。

Reply:
@r0227n 
以下の修正で問題ないでしょうか？

```suggestion
    if [ "$latest_base" != "$base_branch" ] && git show-ref --verify --quiet refs/heads/$base_branch; then
      # 派生元ブランチの最新の共通コミットが 基準ブランチ に取り込まれているか確認
      if git branch --contains $latest_commit | grep -q "$base_branch"; then
        latest_base="$base_branch"

Reply:
@karas1910 
認識あいます！

Reply:
@r0227n
[4c3da3d](https://github.com/yumemi-inc/flutter-mobile-project-template/pull/464/commits/4c3da3d8871f61182ff64aff78df3edf6cf9cef2) で修正しました！